x/leverage/keeper/grpc_query.go

```diff
@@ -541,57 +381,48 @@ func (q Querier) MarketSummary(
 
 	token, err := q.Keeper.GetTokenSettings(ctx, req.Denom)
 	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, "not accepted Token denom")
+		return nil, err
 	}
+
 	rate := q.Keeper.DeriveExchangeRate(ctx, req.Denom)
 	supplyAPY := q.Keeper.DeriveSupplyAPY(ctx, req.Denom)
 	borrowAPY := q.Keeper.DeriveBorrowAPY(ctx, req.Denom)
-	totalSupply, _ := q.Keeper.GetTotalSupply(ctx, req.Denom)
-	availableBorrow := q.Keeper.GetAvailableToBorrow(ctx, req.Denom)
+	supplied, _ := q.Keeper.GetTotalSupply(ctx, req.Denom)
+	balance := q.Keeper.ModuleBalance(ctx, req.Denom)
 	reserved := q.Keeper.GetReserveAmount(ctx, req.Denom)
-	collateral := q.Keeper.GetTotalCollateral(ctx, req.Denom)
+	borrowed := q.Keeper.GetTotalBorrowed(ctx, req.Denom)
 
-	resp := types.QueryMarketSummaryResponse{
-		SymbolDenom:        token.SymbolDenom,
-		Exponent:           token.Exponent,
-		UTokenExchangeRate: rate,
-		Supply_APY:         supplyAPY,
-		Borrow_APY:         borrowAPY,
-		TotalSupplied:      totalSupply.Amount,
-		AvailableBorrow:    availableBorrow,
-		Reserved:           reserved,
-		Collateral:         collateral,
-	}
+	uDenom := q.Keeper.FromTokenToUTokenDenom(ctx, req.Denom)
+	uSupply := q.Keeper.GetUTokenSupply(ctx, uDenom)
+
+	uCollateral := q.Keeper.GetTotalCollateral(ctx, uDenom)
 
+	availableBorrow := q.Keeper.GetAvailableToBorrow(ctx, req.Denom) // TODO #1162 #1163 - update implementation
+
+	resp := types.QueryMarketSummaryResponse{
+		SymbolDenom:            token.SymbolDenom,
+		Exponent:               token.Exponent,
+		UTokenExchangeRate:     rate,
+		Supply_APY:             supplyAPY,
+		Borrow_APY:             borrowAPY,
+		Supplied:               supplied.Amount,
+		Reserved:               reserved,
+		Collateral:             uCollateral,
+		Borrowed:               borrowed.Amount,
+		Liquidity:              balance.Sub(reserved),
+		MaximumBorrow:          supplied.Amount, // TODO #1162 #1163 - implement limits
+		MaximumCollateral:      uSupply.Amount,  // TODO #1163 - implement limits
+		MinimumLiquidity:       sdk.ZeroInt(),   // TODO #1163 - implement limits
+		UTokenSupply:           uSupply.Amount,
+		AvailableBorrow:        availableBorrow,
+		AvailableWithdraw:      uSupply.Amount,                  // TODO #1163 - implement limits
+		AvailableCollateralize: uSupply.Amount.Sub(uCollateral), // TODO #1163 - implement limits
+	}
```

Comment:
Note that the current fields in market summary are all working - only the new ones having to do with safety limits have TODOs.

I would prefer to do them in a separate PR, since they'll require some math and module behavior changes. This PR would meanwhile merge everything proto-breaking, reducing the impact of the implementation PR.